Framework/DebugGUI/src/imgui_internal.h

```diff
@@ -12,10 +12,10 @@
 #error Must include imgui.h before imgui_internal.h
 #endif
 
-#include <stdio.h>      // FILE*
-#include <stdlib.h>     // nullptr, malloc, free, qsort, atoi, atof
-#include <math.h>       // sqrtf, fabsf, fmodf, powf, floorf, ceilf, cosf, sinf
```

Comment:
Did you actually check this works? substituting `math.h` -> `cmath` is not something you can always do mechanically. This is actually why I did not do it... I think we should turn off that clang-tidy check, instead, especially if we apply it to code copied from AliPhysics.

Reply:
I had to fix one problem in a lambda expression after the cmath change.
What was the problem you encountered? Will it be covered by the ctests?

In general I agree that we should not check this code. Actually, I would just move it out of O2, and also avoid the duplicated headers in src and include dir. But this constantly breaks my PRs since I include imgui, so I was hoping this would fix it for now.

Reply:
Agreed on moving it out.

At least in the past including cmath rather than math.h had all weird kind of issues because suddenly abs and co. become templated on the input type. Now, I frankly cannot provide any example quickly, so if it compiles let's go ahead with it, but still I would be cautious with enforcing that change everywhere, especially because we cannot guarantee that `using namespace std` `math.h` `stdlib.h` are not used anywhere in our dependency chain.

Reply:
Hm, but this is a general problem then. In particular, what most headers in o2 will include <cmath>. What happens if there are some headers left that include <math.h>. Will this lead to collisions?
Anyway, the templated version of <cmath> will be in the namespace as soon as <cmath> appears somewhere in the header chain. So I would rather fix the code to be compatible to <cmath>.

Reply:
True, but my point is that we should not propose to use an automatic fix for something which actually requires attention.

Reply:
ok, true, but in this case I was actually going through the files manually, and checked that it compiles.
But OK, I'll merge this when it passes the C.I., so I can go on with the other PRs currently failling the code checker. For the long run, we should factor this out.

Reply:
More on this. It's actually an error to have:

```C++
#include <cmath>

extern "C" {
  // something which uses math.h / cmath functions
}
```

This is because in general it's illegal to have C++ code (due to the templates) in a C section. So again, the automatic fix was wrong. I will propose to next WP3 that we disable that particular check as simply wrong.

Reply:
@ktf : I do not fully understand. The following example compiles for me:
```
#include <cmath>
extern "C" {
  void foo() {
    float a = std::log(10);
  }
}
int main(int, char**) {
  foo();
  return 0;
}
```
basically, what ever you define in the `extern "C"` block will have C linkage, but to my knowledge you are not forbidden to use C++ inside.

One must not include the <cmath> header insider extern "C"
```
extern "C" {
#include <math.h>
```
works, but
```
extern "C" {
#include <cmath>
```
will fail, you have to use
```
#include <cmath>
extern "C" {
```
and this is what I did. (At least I wanted to, If not, I did a mistake.)

Overall, I think we cannot completely avoid this problem anyway. Assume:
**foo.h**:
```
#include <math.h>
extern "C" {
   [something]
}
```
and **bar.cpp**:
```
#include <cmath>
#include "foo.h"
```
The code inside the extern "C" will see <cmath> no matter what.

I agree that we can disable the codechecker warnings for imgui, but I think in general it would be good to have code that is compatible to <cmath>, and I'd prefer to use <cmath> everywhere, in particular not include both <math.h> and <cmath>.

Reply:
```
extern "C" {
#include <cmath>
```

is indeed what is breaking... No idea why it did not spot it in the CI. I guess I will simply move everything out...

Reply:
@ktf : Sorry, then it was my fault and I missed it. In some places I did move it out.

Btw, if you want to be able to still include it in `.c` files, unfortunately you'll have to do:
```
#ifdef __cplusplus
#include <cmath>
extern "C" {
#else
#include <math.h>
#endif
```
(had to do that in `Framework/DebugGUI/src/GL/glcorearb.h`, since it was used in some `.c` file)